addons/crm/models/crm_lead.py

```diff
@@ -1241,6 +1259,65 @@ def _handle_salesmen_assignment(self, user_ids=False, team_id=False):
     # CLASSIFICATION TOOLS
     # --------------------------------------------------
 
+    def _get_potential_lead_duplicates_domain(self, lead, include_lost=False):
+        """ Returns a domain that can be used to retrieve the potential lead duplicates.
+            see: `_get_potential_lead_duplicates`
+            :param lead: Reference lead
+            :param boolean include_lost: If true, the leads considered as lost will be included in the results
+        """
+
+        domain = []
+        for normalized_email in filter(None, [tools.email_normalize(email) for email in tools.email_split(lead.email_from)]):
+            if '@' not in normalized_email:
+                continue
+            _, _, email_domain = normalized_email.rpartition('@')
```

Comment:
We have a email_domain_extract tool method in tools/mail.py ... if rpartition is smarter maybe you could update it accordingly and use it so that we have a single tool instead of doing the same kind of computation in various addons.